Update the border colors in the Cart and Checkout blocks

tarunvijwani opened the pull request:
## What

Fixes #11354

## Why

In this PR, we're updating the border colors in the Cart and Checkout blocks to match the order confirmation block template defaults and create a cohesive experience.

## Testing Instructions

1. Go to the Cart block page and confirm the border color is similar to #E7E7E7.
2. Open the shipping calculator and confirm the form fields have a border color similar to #1E1E1E. 
3. Go to the Checkout block page and confirm the border color is similar to #E7E7E7
4. Confirm the form fields have a border color similar to #1e1e1e. 
5. Test it with different themes different background colors, and confirm that the borders are visible. 

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested (ie: it makes changes to tests, coding standards, docblocks, etc.). -->
* [ ] Should be tested by the development team exclusively <!-- Check this box if this PR should be tested by the development team exclusively (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->


## Screenshots or screencast


https://github.com/woocommerce/woocommerce-blocks/assets/11503784/85714563-e608-474d-8e7b-92fb4351813e





## WooCommerce Visibility

<!-- Check this documentation link (../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WooCommerce core, part of the feature plugin, or experimental. -->
Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:

* [x] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [x] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog
<!-- Provide a brief, descriptive summary of the changes in this PR. Include potential impacts on different parts of the product. Example: "Updated the checkout process to streamline the experience for users and reduce the number of steps." -->

> Update the border colors in the Cart and Checkout blocks.

Commit:
Change border color to Gutenberg 100 for Cart and Checkout blocks

Commit:
Change form input color to Gutenberg 900 for Cart and Checkout blocks

Commit:
Fix additional border colors of Checkout block

Commit:
Update borders and form field colors in Cart Block

tarunvijwani commented:
This PR is on hold. There were concerns about user experience with light border colors. 
Waiting for @elizaan36 to confirm the colors for borders. 

Internal discussion: p1698400336133709/1698387117.246639-slack-C8X6Q7XQU

Commit:
Change border color to $universal-border-light

Commit:
Change border color for cart line items to $universal-border-light

Commit:
Change  form input fields border color to $universal-border-dark

- Add $universal-border-dark color variable for #1e1e1e on white.

Commit:
Change  form input fields border color to $universal-border-dark

- Add $universal-border-dark color variable for #1e1e1e on white.

tarunvijwani commented:
We decided to continue with $universal-border-light color for borders.

Commit:
Fix colors of text area border and form steps

Commit:
Fix form steps

tarunvijwani commented:
Hi @elizaan36, could you take a moment to review the PR? I've included a screen recording for reference. Just wanted to let you know that I added `rgba(0, 0, 0, 0.882)` color for `Gutengerb 900`, which is the same as `#1e1e1e` color. I'd really appreciate your feedback on this.

elizaan36 reviewed (commented):
Hi @tarunvijwani I'm trying to figure out why, but the border color in the implementation is too light and almost disappears on the screen. 

It looks like the correct value from the code, but when I compare it visually to the Figma it's much lighter. 

<img width="1236" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/cfa1fed0-41f9-49fc-a031-437473f6bae8">

Why do you think this is?

Also for the $universal-border-dark variable, could we still use the RGB value for consistency (17, 17, 17, .80) ? Rather than 0/0/0.

tarunvijwani commented:
> It looks like the correct value from the code, but when I compare it visually to the Figma it's much lighter.

@elizaan36 I think that's because the rgba value is not correct. We're using rgba(17, 17, 17, 0.115); which is #e7e7e7 on white. However, on [Figma](https://www.figma.com/file/Xcn9ZirCFwU451IQZxMAcb/Block-based-checkout?node-id=5245%3A75642&mode=dev) we're using #E0E0E0;  Would you be able to help me with the correct rgba value for it?

![image](https://github.com/woocommerce/woocommerce-blocks/assets/11503784/b69618bb-b8dd-4fe6-8ee4-aee8df2b08d4)

Commit:
Update universal dark border rgba value

- Update universal dark border rgba value from rgba(0, 0, 0, 0.882) to rgba(17, 17, 17, .80) to keep it consistent with universal-border colors

elizaan36 commented:
No problem, I'll screenshot what I have now in Figma. For the `$universal-border-light` variable, I'm using `#111111 - 13%`. Which translates to the rgba value (17,17,17,.13). I'm also seeing that the equivalent is #e0e0e0.

<img width="1033" alt="Screenshot 2023-10-31 at 2 24 02 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/e0175841-0a89-419b-ad50-8b11f61d78e0">

Sorry for the confusion there as those Gutenberg values slipped into the mocks by accident. Hopefully this nails it, and I'll spend some time cleaning up the Figma.

One last thing. I noticed that the borders between the section numbers aren't following the same color system. Can we update those borders to the `$universal-border-light` variable as well? It creates a really nice consistency overall on the checkout.

<img width="311" alt="Screenshot 2023-10-31 at 2 26 31 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/4e758ac2-7a83-4cf4-9c36-8153e2805e9a">

Commit:
Update $universal-border-light color

Commit:
Update the opacity of the borders

tarunvijwani commented:
Thank you, @elizaan36, for the color code. 

>One last thing. I noticed that the borders between the section numbers aren't following the same color system. Can we update those borders to the $universal-border-light variable as well? It creates a really nice consistency overall on the checkout.

I noticed form borders had opacity that's why they were appearing more lighter than the actual color. I have now removed the opacity from them.

Here is the updated output: 




https://github.com/woocommerce/woocommerce-blocks/assets/11503784/d6533c13-cff0-44bd-be42-17715654975d

elizaan36 commented:
I think we overcorrected a bit considering the additional transparency you mentioned, so tweaking the color one more time to the lighter rgba (17, 17, 17, .12) or even rgba (17, 17, 17, .11) would work.

Commit:
Update the border color to rgba (17, 17, 17, 0.12)

Commit:
Roll back $universal-border-light value to the original value: 0.115

tarunvijwani commented:
@elizaan36 as we discussed over slack, I have rolled back the $universal-border-light value to rgba(17, 17, 17, 0.115). 

Here is the updated video:

## Storefront theme

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/3edf50fb-a1f3-4175-9b37-04530ca7e52f


## Tove theme

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/5f33aa12-1d7a-47c6-94b2-4bf8a909ff85

elizaan36 commented:
The border color looks perfect now. Thanks, @tarunvijwani 🙏

mikejolley commented on assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss:
```diff
@@ -52,11 +52,12 @@ $border-width: 1px;
 			height: $universal-border-radius - $border-width;
 			margin-right: $gap-small;
-			opacity: 0.3;
+			opacity: 1;
```
Are these values needed? 1 should be default.

tarunvijwani replied:
@mikejolley While testing the PR, there were some instances where opacity was overridden to 0.3 again. I added this manually to make sure it remains the same and also doesn't inherit the theme opacity styles.

mikejolley replied:
I think you should investigate where its coming from and fix there. It could be a mixin like `with-translucent-border`.

tarunvijwani replied:
@mikejolley Thank you for the suggestion! I have passed opacity argument in the `with-translucent-border` as it accepts opacity and assign it to the psuedo element: 

```css

@mixin with-translucent-border($border-width: 1px, $opacity: 0.3) {
	position: relative;

	&::after {
		border-style: solid;
		border-width: $border-width;
		bottom: 0;
		content: "";
		display: block;
		left: 0;
		opacity: $opacity;
		pointer-events: none;
		position: absolute;
		right: 0;
		top: 0;
	}
}
```

mikejolley commented:
Moving to 11.6.0

Commit:
Remove Opacity and pass it as argument in with-translucent-border

- We have with-translucent-border option that accepts border and opacity for pseudo elements.

Commit:
Remove additional spaces

Commit:
Replace with-translucent-border with normal border

- Since we're are not mixing the opacity to the border color so we don't need to use mixin with-translucent-border.

tarunvijwani commented:
@mikejolley Thank you for the suggestions, I have replaced `with-translucent-border` with normal borders. Please take a moment to review the PR again. 🙂

nielslange reviewed (commented):
Thanks for working on this, @tarunvijwani. I quickly checked the PR against the Twenty Twenty-Four theme and some borders are pretty hard to read. 

## Style: Default

<table>
<tr>
<td valign="top">Before:
<br><br>
<img width="1278" alt="Screenshot 2023-11-08 at 16 37 46" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/053b9a49-d01d-4d99-bc42-a3812cdc7b51">
</td>
<td valign="top">After:
<br><br>
<img width="1278" alt="Screenshot 2023-11-08 at 16 41 44" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/b2449bd3-19cf-4535-83a9-a9dfe02d1150">
</td>
</tr>
</table>

## Style: Ember

<table>
<tr>
<td valign="top">Before:
<br><br>
<img width="1280" alt="Screenshot 2023-11-08 at 16 46 40" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/d410f0fd-354a-4f1b-aeb7-1ce9d672a7ba">
</td>
<td valign="top">After:
<br><br>
<img width="1279" alt="Screenshot 2023-11-08 at 16 41 31" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/1f204013-5dc7-4f86-af4d-0925d3ecd483">
</td>
</tr>
</table>

## Style: Maelstrom

<table>
<tr>
<td valign="top">Before:
<br><br>
<img width="1278" alt="Screenshot 2023-11-08 at 16 38 34" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/c9ecead7-758f-443f-8586-ba7c655139dd">
</td>
<td valign="top">After:
<br><br>
<img width="1277" alt="Screenshot 2023-11-08 at 16 41 18" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/e1856a6f-7f0d-4bb2-be89-4c6814360c2a">
</td>
</tr>
</table>

mikejolley commented:
@nielslange I don't think we should get too caught up in supporting every possible combination because we're trying to create a generic solution that works in most cases. There will clearly be borders that need to be set by the merchant when creating the blocks with global styles. 

Do you have thoughts on this @ralucaStan?

mikejolley commented on assets/css/abstracts/_colors.scss:
```diff
@@ -20,3 +20,4 @@ $image-placeholder-border-color: #f2f2f2;
 $universal-border: rgba(17, 17, 17, 0.3); // Used for form step borders.
 $universal-border-light: rgba(17, 17, 17, 0.115); // e7e7e7 on white.
 $universal-body-low-emphasis: rgba(17, 17, 17, 0.5); // Used for low emphasis text such as input labels.
+$universal-border-dark: rgba(17, 17, 17, 0.8); // #1e1e1e on white. Used for input borders.
```
This is fine for now but we may need to revisit naming depending on if it makes sense to target elements (-input-border) vs shades (-border-dark).

mikejolley commented on assets/js/base/components/cart-checkout/order-summary/style.scss:
```diff
@@ -27,6 +27,8 @@
 		}
 
 		&:last-child {
+			border-width: 0;
```
Is this to reset the style above? I would prefer to see `border-bottom: 0` if so, just resetting what you added before.

mikejolley commented on assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss:
```diff
@@ -81,12 +81,12 @@ $border-width: 1px;
 
 	.wc-block-components-express-payment__content {
-		@include with-translucent-border(0 $border-width $border-width);
+		border-style: solid;
+		border-width: 0 $border-width $border-width;
+		bottom: 0;
+		border-color: $universal-border-light;
```
I'm not sure what you prefer, but if it is more succinct this could be written with:

```suggestion
		border: $border-width solid $universal-border-light;
		border-top: 0;
		bottom: 0;
```

mikejolley commented on assets/js/blocks/cart-checkout-shared/payment-methods/style.scss:
```diff
@@ -198,7 +198,11 @@
 	.wc-block-components-radio-control-accordion-option,
 	.wc-block-components-radio-control__option {
-		@include with-translucent-border(1px 1px 0 1px);
+		border: 1px solid $universal-border-light;
```
This one had no bottom border previously—was this change intentional?

tarunvijwani replied:
Yes, this change was intentional. Previously, the borders were added on pseudo elements.

mikejolley reviewed (commented):
There are a few more instances of `$input-border-gray`. Should we replace those and remove the style rule so its not used going forward?

nielslange commented:
> @nielslange I don't think we should get too caught up in supporting every possible combination because we're trying to create a generic solution that works in most cases. [...]

It's not my intension to find a solution for each case. I simply wanted to point out that the current solution has a very low color contrast ratio. IMO, these colors are far from a generic solution as some borders are very hard to see that way.

nielslange commented:
> @nielslange I don't think we should get too caught up in supporting every possible combination because we're trying to create a generic solution that works in most cases. [...]

@mikejolley It's not my intension to find a solution for each case. I simply wanted to point out that the current solution has a very low color contrast ratio. IMO, these colors are far from a generic solution as some borders are very hard to see that way. A generic solution, should work for the majority of users, independent of certain conditions, such as eye disorders. 

https://wordpress.org/about/accessibility/ states:

> WordPress aims to make the WordPress Admin and bundled themes fully WCAG 2.0 AA compliant where possible.

Regarding the color contrast, https://make.wordpress.org/accessibility/handbook/design/use-of-color/#color-contrast states:

> - 4.5:1 for normal text
> - 3:1 for [[large text (24px equivalent or 19px equivalent and bold)](http://mcdpartners.com/lab/meeting-wcag-color-contrast-guideline/)](http://mcdpartners.com/lab/meeting-wcag-color-contrast-guideline/)

Using the default style of the Twenty Twenty-Four theme, without applying any modifications, results in [a color contrast ratio of 1.27:1](https://coolors.co/contrast-checker/dfdfdf-f9f9f9).

<img width="1171" alt="Screenshot 2023-11-08 at 20 54 07" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/c35cf5c2-1860-4044-80ab-5b6652dd8e33">

My primary intension is to prevent excluding folks with vision impairment from using WooCommerce Blocks due to accessibility reasons.

mikejolley commented:
@nielslange that resource refers to text though doesn't it. It's my understanding that this PR and the designs in Figma only adjust border colors.

nielslange commented:
> @nielslange that resource refers to text though doesn't it. It's my understanding that this PR and the designs in Figma only adjust border colors.

It's true that this PR only affects borders. That said, I consider borders as visual elements that should be visible. The moment their contrast ratio is too low, that's the moment when these visual elements start to become invisible for visual impaired users. I do not want to block this PR, though. I only want to raise my concerns that by applying these border colors we will exclude certain people from seeing them.

ralucaStan commented:
@nielslange what borders are failing the a11y check? If it is only the decoration ones used for visual separation those can be ignored IMO. They only have a decorative purpose

nielslange commented:
As I said, I don't want to block this PR. I only want to state that this way forward will affect visually impaired people, @ralucaStan .

ralucaStan commented:
@nielslange I understand and appreciate the intention. My question still holds. Which borders are failing the a11y check? Because if the form elements borders are failing that might lead to some people not being able to fill in the form. The decoration borders don't have an impact in the checkout process.

nielslange commented:
> @nielslange I understand and appreciate the intention. My question still holds. Which borders are failing the a11y check? Because if the form elements borders are failing that might lead to some people not being able to fill in the form. The decoration borders don't have an impact in the checkout process.

Currently, only decorative borders fail the accessibility check. However, I do not understand why we have decorative borders, knowing that they will not be visible to everyone. IMO, we should either have borders that are visible to all users, independent of visual impairment, or no borders at all. Having borders, that are only visible to non-visually impaired users, sounds very exclusive to me. Personally, I'm not visually impaired, yet I already have a problem in seeing the decorative borders on the following screenshot, which shows the Twenty Twenty-Four theme without any adjustments:

<img width="1206" alt="Screenshot 2023-11-08 at 22 47 02" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/62903e4e-8971-41f6-81cb-90ff91ba1bd4">

As mentioned before, it's not my intension to block this PR in any way. If you feel that we should do that, then please go ahead. Personally, I doubt that this should be the way forward, as we do exclude certain users from seeing specific elements, even if we only consider them as decorative. My primary intension is that the UI of WooCommerce Blocks is as inclusive as possible. IMO, this PR is rather exclusive than inclusive.

mikejolley commented:
@nielslange I understand the point you're trying to put across but saying stuff like this you're basically saying that if we merge this we don't care about users with impaired vision? This isn't really true. 

> If you feel that we should do that, then please go ahead. Personally, I doubt that this should be the way forward, as we do exclude certain users from seeing specific elements, even if we only consider them as decorative.

The accessibility docs apply to text. @ralucaStan was correct to also point out that important non-decorative elements should also have sufficient contrast (input borders) which you've confirmed is the case. It looks like the designs in Figma took important/decorative borders into account already.

> However, I do not understand why we have decorative borders, knowing that they will not be visible to everyone. IMO, we should either have borders that are visible to all users, independent of visual impairment, or no borders at all. 

By this logic everything would be plain text, otherwise we'd exclude users of screen readers too :)

I'd vote to proceed with the change. @tarunvijwani maybe that would be good to document alongside the definitions of the universal colors; ensure that the low contrast colors are marked as used for decorative elements only so its not applied to inputs or text by accident in the future.

ralucaStan commented:
> I'd vote to proceed with the change. @tarunvijwani maybe that would be good to document alongside the definitions of the universal colors; ensure that the low contrast colors are marked as used for decorative elements only so its not applied to inputs or text by accident in the future.

Sounds good to me. Thank you all for your input. I appreciate the care. Should we see any impact of this PR, as always, we'll be quick to act on it.

tarunvijwani commented:
@nielslange, thank you for taking the time to comment. 
I appreciate your concerns about the border colors, and I have a potential solution that we can consider. We can add an option for the merchants to change the color of the decorative borders. This way, we'll be aligned with FSE and provide more control to the merchants in styling their sites. Currently, they don't have this option, and I agree that it's not a good user experience. However, I understand that adding this feature would require extra effort. We can consider opening a new ticket for it and work on it in a later cycle. 
@ralucaStan @mikejolley Please let me know your thoughts on this.

tarunvijwani commented:
>@tarunvijwani maybe that would be good to document alongside the definitions of the universal colors; ensure that the low contrast colors are marked as used for decorative elements only so its not applied to inputs or text by accident in the future.

@mikejolley That's a great suggestion. I'll document the definitions. 👍

ralucaStan commented:
@tarunvijwani does not hurt to open a new issue, but make it a low priority. Thank you

Commit:
Fix applied css and add low contrast Color usage details

-  Fix minor CSS as per the standards.
- Add comments for universal border colors that they're low contrast colors and should be used for decorative elements only

tarunvijwani commented:
@mikejolley I have made all the suggested changes. Please take a moment to review the PR again. Thank you so much!

mikejolley reviewed (changes requested):
Code changes look good and I reviewed the cart/checkout flow and it seems to be working as intended. I noticed one issue that needs resolving. There are gaps on the top borders in cart. See:

![Screenshot 2023-11-14 at 10 59 23](https://github.com/woocommerce/woocommerce-blocks/assets/90977/b4ab03c4-ab0e-47d0-bd72-fb1dccbf8fe0)

Commit:
Fix border gap and double border for multiple shipping packages

tarunvijwani commented:
@mikejolley Fixed border gap in https://github.com/woocommerce/woocommerce-blocks/pull/11474/commits/f1603e74cec083fafd1b152e5053f2524c82cc0e 👍

mikejolley commented:
@tarunvijwani now there is no top border at all:
![Screenshot 2023-11-15 at 14 18 02](https://github.com/woocommerce/woocommerce-blocks/assets/90977/39cb148c-206f-43c7-ae28-6b77de97527a)

ralucaStan commented:
@tarunvijwani this PR is still using the old PR template so I wanted to make sure you tested these UI changes on Firefox, Safari as well and on mobile

mikejolley commented:
@tarunvijwani I checked out the latest changes and the top border has now vanished in editor and frontend. This needs further testing?

![Screenshot 2023-11-15 at 14 18 02](https://github.com/woocommerce/woocommerce-blocks/assets/90977/07b99b26-fc0e-4035-a520-6cd14cdb21e7)

Commit:
Fix payment method borders and Cart line items borders

- Change border bottom to border-top for cart line items.
- Fix payment method radio control borders.

Commit:
Fix local pickup border

- Local pickup was using  with-translucent-borders so it has some opacity, changed it to border-bottom.

Commit:
Add border-bottom to cart items

Commit:
Remove additional border from Cart items

tarunvijwani commented:
> @tarunvijwani this PR is still using the old PR template so I wanted to make sure you tested these UI changes on Firefox, Safari as well and on mobile

Yes, I have tested it on different browsers and screen sizes. I have updated the template of the PR as well. Thank you, @ralucaStan! 

>@tarunvijwani I checked out the latest changes and the top border has now vanished in editor and frontend. This needs further testing?

@mikejolley I have fixed the borders now. Here is the screen recording of the final version(Updated the video the PR description): 

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/85714563-e608-474d-8e7b-92fb4351813e

mikejolley commented:
Checkboxes are using the wrong color @tarunvijwani 

![Screenshot 2023-11-16 at 12 16 08](https://github.com/woocommerce/woocommerce-blocks/assets/90977/ffe90df2-dcf5-4bdf-bc96-7931d25795b4)

mikejolley reviewed (changes requested):
Checkboxes are using the wrong color @tarunvijwani 

![Screenshot 2023-11-16 at 12 16 08](https://github.com/woocommerce/woocommerce-blocks/assets/90977/ffe90df2-dcf5-4bdf-bc96-7931d25795b4)

I would review Figma before marking this ready to review again to ensure its matching.

Commit:
Fix radio and checkbox borders as per the design

Commit:
Fix shipping method borders and background color as per the design.

Commit:
Force align left on the description for the local pickup options.

tarunvijwani commented on assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/style.scss:
```diff
@@ -50,7 +50,7 @@
 	.wc-block-components-radio-control__secondary-description {
 		width: 100%;
-		text-align: left;
+		text-align: left !important; // To force align left on the description for the local pickup options.
```
On trunk, the local pick-up description was aligned right.  Added `!important` to force align left.

| Before | After |
|---|---|
|![image](https://github.com/woocommerce/woocommerce-blocks/assets/11503784/817776a7-366a-4b9b-b036-b3666d583f87)|![image](https://github.com/woocommerce/woocommerce-blocks/assets/11503784/0d8d061b-967d-48ae-ba09-215bdd73366e)|

tarunvijwani replied:
Reverted the change as it got fixed in trunk

tarunvijwani commented:
@elizaan36 Couple of questions related to latest Figma design: Xcn9ZirCFwU451IQZxMAcb-fi-4749%3A86624:

## 1. Checkout Block: Shipping method background color:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/11503784/73b6ed5f-814c-4840-982b-f31d04b02555)

In the design, there are a couple of mockups, and a few of them do not have any background color for the shipping method that is not selected. Can you confirm that there will be no color for the unselected shipping method?


## 2. Checkout Block: Accent color focus shadow:

Should we add accent color focus shadow for the selected text input?

![image](https://github.com/woocommerce/woocommerce-blocks/assets/11503784/66f8dae3-260f-4807-b4b6-bff27ba317a4)


## 3. Cart Block:  Quantity Selector
![image](https://github.com/woocommerce/woocommerce-blocks/assets/11503784/a774969a-339d-47d2-a977-3bdc152776d3)
The border color for the quantity is not matching with the overall border color. Please confirm the border color for quantity. I believe it should match with all the borders to keep it consistent.
Example: 

<img width="793" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/11503784/cfb9f617-9a72-433c-890b-53867deae665">

elizaan36 commented:
> In the design, there are a couple of mockups, and a few of them do not have any background color for the shipping method that is not selected. Can you confirm that there will be no color for the unselected shipping method?

@tarunvijwani In `pdFofs-1dB#comment-1975` I shared details on the interaction of the shipping method section in a video. The background appears on hover and active states only, there's no background color for the unselected shipping method.

> Should we add accent color focus shadow for the selected text input?

Yes! We can include this as a future enhancement in another issue, I think it would improve accessibility and personality but needs to be tested.

> Please confirm the border color for quantity. I believe it should match with all the borders to keep it consistent.
Example:

Good call out and your example looks great. I updated the Figma to show the $universal-border-light variable for the Cart and quantity picker outline. 
<img width="1066" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/8223a8f2-129e-46ff-86b6-7227ab720486">

Commit:
Merge remote-tracking branch 'origin/trunk' into update/11354-cart-and-checkout-blocks-colors

Commit:
Update border color in quantity selector component

ralucaStan commented:
@tarunvijwani if this PR is still in progress then let's make it a draft until it's good for review again.

tarunvijwani commented:
@elizaan36 As discussed, I'm converting the PR to draft till we finalize the design. 

Here is the link to the JN sites for your reference: 
This PR:  https://bare-spider.jurassic.ninja/
WC Blocks 11.6.1 version: https://serious-scarlet.jurassic.ninja/

elizaan36 commented:
Thanks @tarunvijwani All looks good, a couple notes.

I discussed this with the design team and we need to define a medium gray for the resting state of both radio buttons and checkboxes because the light gray looks disabled and the strong border color is too noisy for these selectors.

So overall we have three grays:
- $universal-border-light:rgba (17, 17, 17, .11); For section borders
- $universal-border-medium:rgba (25, 23, 17, .3); For radio and checkbox resting state
- $universal-border-strong: rgba(17, 17, 17, 0.8); For input borders.

<img width="2526" alt="checkout clean up" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/59ad6d0d-e1d9-4eb6-b38e-59099a1304b3">

Also, in the prototype it looks like the radio buttons aren't horizontally aligned. The shipping and payment method selections should appear in the same vertical row.

<img width="345" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/7b6fecc6-595c-4fe0-99b5-13278b3cf07a">

Figma: Xcn9ZirCFwU451IQZxMAcb-fi

Commit:
Merge branch 'trunk' into update/11354-cart-and-checkout-blocks-colors

Commit:
Fix Shipping options radio selection alignment

- Add left padding to wc-block-components-shipping-rates-control__package element.

Commit:
Update colors and variables

tarunvijwani commented:
Thank you, @elizaan36, for sharing the final design. I have incorporated all the changes in this PR and created a JN site for you to have a look: https://apparent-orca.jurassic.ninja/. Please take a moment to review the site and let me know if any additional changes are required. 

For `$universal-background` color, I kept it `rgba(17, 17, 17, 0.02)` as `rgba(17, 17, 17, 0.2)` was appearing too dark.

Here is the screenshot for `rgba(17, 17, 17, 0.2)`:


<img width="890" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/11503784/e5be9092-208a-4514-a098-542190d361ed">

elizaan36 commented:
Looks great! 🙏
You were right about the color. I mistakenly wrote .2 but 2% = .02.

Commit:
Merge branch 'trunk' into update/11354-cart-and-checkout-blocks-colors

tarunvijwani commented:
@elizaan36 Thank you for the confirmation. 
@nielslange @mikejolley The PR is ready for review. 🙂

nielslange reviewed (commented):
Thanks for working on this, @tarunvijwani. I've been reviewing the contrast aspects in line with our discussions and the WCAG guidelines. While the [Text Contrast guidelines](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) focus on text elements, the [Non-Text Contrast guidelines](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) also emphasize the importance of sufficient contrast for UI components, including checkboxes and radio buttons.

I noticed that in the default styles of the Twenty Twenty-Four theme, checkboxes and unselected radio buttons have a contrast ratio of 1.86:1, which is below the recommended 3:1 ratio. This could impact users with visual impairments. These are the UI elements I'm referring to:

<img width="711" alt="Screenshot 2023-12-06 at 16 10 41" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/435a633e-5737-4850-b8b5-3e3c211348fc">

Would it be possible to adjust these elements to align with the 3:1 contrast ratio as suggested by the guidelines? This enhancement could significantly improve accessibility for all users.

cc: @elizaan36

tarunvijwani commented:
Increasing the contrast of the border color for the checkbox and radio buttons could lead to an inconsistent look. However, we should consider the WCAG guidelines. I will wait for @elizaan36 to provide her suggestions.

elizaan36 commented:
We can bump the $universal-border-medium to `rgba (25, 23, 17, .48)`

The increase in opacity passes in my contrast checker, let me know if that passes on your end.

nielslange commented:
> We can bump the $universal-border-medium to `rgba (25, 23, 17, .48)`
> 
> The increase in opacity passes in my contrast checker, let me know if that passes on your end.

Thanks for considering this change, @elizaan36. I want to clarify that my intention isn't to block this PR. I'm simply very passionate about a11y and strive to ensure that all UI elements are equally accessible.

Commit:
Change checkbox and radio button colors

- Change checkbox and radio button colors from  rgba(25, 23, 17, 0.3) to  rgba(25, 23, 17, 0.48).

tarunvijwani commented:
Thank you, @nielslange and @elizaan36! I have updated the colors, here is how it looks like on the Twenty Twenty Four theme:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/11503784/88aa7cfd-d915-4c0e-a2b4-82b47df167f0)

nielslange commented on assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss:
```diff
@@ -81,12 +81,11 @@ $border-width: 1px;
-		@include with-translucent-border(0 $border-width $border-width);
+		border: $border-width solid $universal-border-light;
+		border-top: 0;
+		bottom: 0;
```
`bottom: 0;` does not seem to have any effect here:

<img width="346" alt="Screenshot 2023-12-07 at 15 28 58" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/ce49be52-df8e-4494-b7fd-a7528e79eaba">

tarunvijwani replied:
@nielslange Nice catch! Thank you, I have removed it.

nielslange replied:
Thanks for adjusting this part.

nielslange reviewed (commented):
Thanks for working on this, @tarunvijwani. Overall, the PR looks good. I just noticed one definition that doesn't seem to do anything.

Commit:
Remove bottom property from express payment style

nielslange reviewed (approved):
Looking great now, @tarunvijwani. Let's ⛴️ this change. 🙌

elizaan36 reviewed (commented):
LGTM! Thanks!

Commit:
Merge branch 'trunk' into update/11354-cart-and-checkout-blocks-colors
